homework-03/utils/streams.js

```diff
@@ -0,0 +1,119 @@
+const commander = require('commander')
+const trough = require('through2')
+const fs = require('fs')
+const syspath = require('path')
+const csvjson = require('csvjson')
+
+function reverse() {
+  process.stdin.setEncoding('utf8')
+
+  process.stdin.on('readable', () => {
+    const chunk = process.stdin.read()
+    if (chunk !== null) {
+      process.stdout.write(`data: ${chunk}`)
+    }
+  })
+
+  process.stdin.on('end', () => {
+    process.stdout.write('end')
+  })
+}
+
+function transform() {
+  const transformStream = trough(write, end);
+  process.stdin.pipe(transformStream).pipe(process.stdout);
+}
+
+function write (buffer, encoding, next) {
+  this.push(buffer.toString().toUpperCase());
+  next();
+}
+
+function end (done) {
+  done();
+}
+
+function outputFile (filePath) {
+  if (fs.existsSync(filePath)) {
+    const stream = fs.createReadStream(filePath)
+    stream.pipe(process.stdout);
+  } else {
+    printError()
+  }
+}
+
+function convertFromFile(filePath) {
+  if (fs.existsSync(filePath) && filePath.endsWith('.csv')){
+    const read = fs.createReadStream(filePath)
+    const toObject = csvjson.stream.toObject()
+    const stringify = csvjson.stream.stringify()
+    read.pipe(toObject).pipe(stringify).pipe(process.stdout)
+  } else {
+    printError()
+  }
+}
+
+function convertToFile(filePath) {
+  if (fs.existsSync(filePath) && filePath.endsWith('.csv')){
+    const read = fs.createReadStream(filePath)
+    const write = fs.createWriteStream(filePath.replace('.csv', '.json'))
+    const toObject = csvjson.stream.toObject()
+    const stringify = csvjson.stream.stringify()
+    read.pipe(toObject).pipe(stringify).pipe(write)
+  } else {
+    printError()
+  }
+}
+
+function cssBundler(path) {
+  if (fs.existsSync(path)) {
+    fs.readdir(path, function (err, files) {
+      if (err) {
+        console.log('Error reading directory ' + path)
+        console.log(err)
+        return
+      }
+      const write = fs.createWriteStream(syspath.join(path, 'bundle.css'))
+      for (let file of files) {
+        let data = fs.readFileSync(syspath.join(path, file), { encoding: 'utf8' })
+        write.write(data)
+      }
+
+      let finalData = fs.readFileSync('nodejs-homework3.css', { encoding: 'utf8' })
```

Comment:
use `path` package here, because if I start this file from folder different that root, conversion doesn't work at all and I don't see any explaining message

Reply:
fixed